File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        int ans = 0;
        map<int,int> res;
        for(auto &i : tasks ){
            res[i]++;
        }
        for(auto [i,j] : res){
            if (j <2) return -1;
            if(j%3 == 0) ans += j/3;
            else  ans+= j/3+1;
        }
        return ans;
    }
};

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        ans = 0
        res = {}
        for i in tasks :
            if i in res :
                res[i] += 1
            else :
                res[i] = 1 
        
        for i,j in res.items():
            if j < 2 :
                return -1
            while(res[i] > 0):
                if  res[i]%3 == 0 :
                    res[i] -=3 
                else :
                    res[i] -=2
                ans +=1
        return ans 